p2p/src/network.rs

```diff
@@ -198,13 +198,17 @@ where
     type Result = ();
 
     async fn handle(&mut self, msg: ConnectPeer) {
+        if self.peers.contains_key(&msg.peer.public_key) {
```

Comment:
Style point, I usually prefer having a matching `else`.

Reply:
I like it like this when it is a pure early return. As opposed to a different code path.